parquet/tests/arrow_reader/mod.rs

```diff
@@ -987,6 +991,33 @@ fn create_data_batch(scenario: Scenario) -> Vec<RecordBatch> {
                 make_utf8_batch(vec![Some("e"), Some("f"), Some("g"), Some("h"), Some("i")]),
             ]
         }
+        Scenario::TruncatedUTF8 => {
+            // Make utf8 batch with strings longer than 64 bytes
+            // to check truncation of row group statistics
+            vec![
+                make_utf8_batch(vec![
+                    Some(&("a".repeat(64) + "1")),
+                    Some(&("b".repeat(64) + "2")),
+                    Some(&("c".repeat(64) + "3")),
```

Comment:
```suggestion
                    None,
```
Try throwing a null in the first batch, that seems to clear up the issue with empty string showing up in the stats. It seems like the schema is inferred from the first batch, so all values are not nullable. Putting a null in the first batch fixes that.

Reply:
Sounds reasonable for me. But I still wonder if its as expected that a first batch without null values will cause the `None`s in following batches are converted into empty strings.

Reply:
I'll have to study the API used here in more detail, but I would assume one would usually provide a schema prior to writing batches. The issue here AFAICT is that a null value is written, but the schema says values are not nullable, so the null is converted to an empty string. I think I'd prefer an error be thrown in this instance.

Reply:
Ahh, with sleep I now see the schema is set in `make_test_file_rg`...the first batch is used.
https://github.com/apache/arrow-rs/blob/0ae9f66d10141c8d5054fd77f73168c7a2ea2819/parquet/tests/arrow_reader/mod.rs#L1037-L1039
Perhaps a note should be added that if nulls are desired, they need to be included in the first batch.